.github/workflows/release.yml

```diff
@@ -132,7 +161,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Download release artifacts
-        uses: actions/download-artifact@v2
+        uses: actions/download-artifact@v4
```

Comment:
[Artifacts uploaded by v3 cannot be downloaded by v4.](https://github.com/actions/download-artifact#breaking-changes) The release workflow should always use v3 for the artifact actions.

Reply:
Good find. This seems like a problem to me. IIUC the deprecation notice correctly, at some point Node 16 actions will not be able to run on the standard GitHub Actions runners. Thus we'd have to use v4, and could no longer download the manylinux artifacts uploaded by v3.

One workaround would be to run this job that downloads and tests the release artifacts inside of the manylinux image for the manylinux artifacts. But at some point it might be easier to simply migrate the manylinux builds to a separate CI pipeline (either its own dedicated GitHub Actions pipeline(s) or switch to another service like Azure).